parcels/particleset.py

```diff
@@ -69,22 +69,25 @@ class ParticleSet(object):
     Please note that this currently only supports fixed size particle
     sets.
 
-    :param size: Initial size of particle set
     :param grid: Grid object from which to sample velocity
     :param pclass: Optional class object that defines custom particle
-    :param lon: List of initial longitude values for particles
-    :param lat: List of initial latitude values for particles
+    :param lon: Optional list of initial longitude values for particles
+    :param lat: Optional list of initial latitude values for particles
     :param start: Optional starting point for initilisation of particles
                  on a straight line. Use start/finish instead of lat/lon.
     :param finish: Optional end point for initilisation of particles on a
                  straight line. Use start/finish instead of lat/lon.
     :param start_field: Optional field for initialising particles stochastically
                  according to the presented density field. Use instead of lat/lon.
+    :param size: Optional initial size of particle set, only required when using
+                 start/finish or start_field arguments
```

Comment:
I wonder if we ultimately want to create separate constructor methods, similar to the ones we have for grids/fields. Maybe something like `ParticleSet.from_array(lon, lat)` and `ParticleSet.from_line(size, lon, lat)`? But definitely not required for this PR, just a thought really.